packages/playground/src/utils/validators.ts

```diff
@@ -777,3 +777,20 @@ export async function isAvailableName(grid: GridClient, name: string) {
     return { message: "Name is already taken." };
   }
 }
+
+/**
+ * Validates an SMTP input string.
+ *
+ * Checks if the input string is a valid email address and does not contain special characters.
+ *
+ * @param {string} input - The input string to validate.
+ * @returns {{ message: string }} - An object with an error message if the input is invalid, or undefined if the input is valid.
+ */
+
+export function isValidSmtp(input: string) {
```

Comment:
1- where is the return type of the function?
2-if the function is named `isX` is expect it to return a `boolean`.  True if X and false if not. To return null that may denote a failure e.g failure to allocate memory, or no data available in such context.

So we have multiple options

1- return a boolean (and discard the error messages, keep it constant to invalid smtp information) 
2- return a boolean with a context, e.g `boolean, null` in case of success and `boolean and error string` in case of failure `  so [boolean, string|null]
3- return a boolean (success) or an error message (failure) `boolean | string`
4- return a new type e.g Result that encapsulates the success/failure status, and message/error fields

```
type Result<T> = {
  success: true;
  value: T;
} | {
  success: false;
  error: string;
};
```

Reply:
good point. the function is deleted now and the validation created in the input rules itself.